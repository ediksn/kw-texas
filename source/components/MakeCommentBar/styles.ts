import { StyleSheet } from 'react-native'
import { verticalScale, scale } from 'react-native-size-matters'
import { theme } from '~/constants'

export const styles = StyleSheet.create({
  wrapper: {
    backgroundColor: theme.backgroundWhite
  },
  avatar: {
    aspectRatio: 1,
    width: scale(32),
    height: scale(32),
    borderRadius: 30
  },
  avatarDefault: {
    width: scale(30),
    height: scale(30),
    opacity: 0.5
  },
  makeCommentBar: {
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
    backgroundColor: theme.backgroundWhite,
    paddingVertical: verticalScale(10),
    paddingHorizontal: scale(10)
  },
  makeCommentBarAvatar: {
    paddingRight: scale(10)
  },
  input: {
    paddingVertical: verticalScale(8),
    paddingHorizontal: scale(5),
    borderRadius: scale(4),
    width: '100%',
    flex: 1,
    backgroundColor: theme.greenLightColor
  },
  sendIconButton: {
    justifyContent: 'center',
    alignItems: 'center',
    paddingLeft: scale(10)
  },
  sendIconButtonFixed: {
    alignItems: 'flex-end',
    paddingTop: verticalScale(4),
    paddingBottom: verticalScale(10),
    paddingHorizontal: scale(10)
  }
})
