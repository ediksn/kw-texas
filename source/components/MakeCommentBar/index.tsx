import React, { useState } from 'react'
import { Keyboard, TextInput, TouchableOpacity, View } from 'react-native'
import { useTranslation } from 'react-i18next'
import { styles } from './styles'
import Avatar from '~/components/Avatar'
import { theme } from '~/constants'
import { Icon } from '~/components'
import { MAX_CHARACTERS_NEW_POST } from '~/utils/constants'

interface MakeCommentsBarProps {
  usrData: any
  onSubmitComment: (comment: string) => void
  keyboardOpen: boolean
}

const MakeCommentBar = ({ usrData, onSubmitComment, keyboardOpen }: MakeCommentsBarProps) => {
  const [comment, setComment] = useState('')
  const { t } = useTranslation()

  const onSubmitCommentInternal = () => {
    onSubmitComment(comment)
    setComment('')
    Keyboard.dismiss()
  }
  const SendIconButton = () => (
    <TouchableOpacity onPress={onSubmitCommentInternal}>
      <Icon name='send-icon' size={28} color={theme.post.green} />
    </TouchableOpacity>
  )

  return (
    <View style={styles.wrapper}>
      <View style={styles.makeCommentBar}>
        <View style={styles.makeCommentBarAvatar}>
          <Avatar
            uri={
              usrData?.photo || typeof usrData?.photo !== 'undefined'
                ? `https://avatar.kwconnect.com/${usrData?.assigneeId}.jpeg`
                : ''
            }
            avatarStyle={styles.avatar}
            avatarDefaultStyle={styles.avatarDefault}
          />
        </View>
        <TextInput
          value={comment}
          style={styles.input}
          maxLength={MAX_CHARACTERS_NEW_POST}
          textAlignVertical='top'
          placeholder={t('components_MakeCommentBar_Your_comment')}
          multiline
          placeholderTextColor={theme.greenColor}
          onChangeText={setComment}
        />
        {!keyboardOpen && (
          <View style={styles.sendIconButton}>
            <SendIconButton />
          </View>
        )}
      </View>
      {keyboardOpen && (
        <View style={styles.sendIconButtonFixed}>
          <SendIconButton />
        </View>
      )}
    </View>
  )
}

export default MakeCommentBar
