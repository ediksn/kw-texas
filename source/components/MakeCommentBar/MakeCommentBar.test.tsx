import React from 'react'
import { render, RenderAPI } from '@testing-library/react-native'
import MakeCommentBar from '~/components/MakeCommentBar/index'

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: any) => key })
}))

jest.mock('react-redux', () => ({
  useSelector: () => ({})
}))

jest.mock(
  'rn-fetch-blob',
  () => {
    return {
      DocumentDir: () => {},
      ImageCache: {
        get: {
          clear: () => {}
        }
      },
      fs: {
        exists: jest.fn(),
        dirs: {
          MainBundleDir: () => {},
          CacheDir: () => {},
          DocumentDir: () => {}
        }
      }
    }
  },
  { virtual: true }
)

describe('Render Make Comment Bar Test', () => {
  let component: RenderAPI

  beforeEach(() => {
    component = render(<MakeCommentBar onSubmitComment={() => ({})} keyboardOpen={false} usrData={null} />)
  })

  it('Match with the snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('Renders correctly', () => {
    expect(component).toBeDefined()
  })
})
