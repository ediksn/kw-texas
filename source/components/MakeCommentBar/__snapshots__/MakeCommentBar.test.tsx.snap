// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Make Comment Bar Test Match with the snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingHorizontal": 21.428571428571427,
        "paddingVertical": 19.61764705882353,
      }
    }
  >
    <View
      style={
        Object {
          "paddingRight": 21.428571428571427,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Array [
            Object {
              "height": 62.857142857142854,
              "opacity": 0.5,
              "width": 62.857142857142854,
            },
            Object {
              "height": 64.28571428571428,
              "opacity": 0.5,
              "width": 64.28571428571428,
            },
          ]
        }
      />
    </View>
    <TextInput
      allowFontScaling={true}
      maxLength={500}
      multiline={true}
      onChangeText={[Function]}
      placeholder="components_MakeCommentBar_Your_comment"
      placeholderTextColor="#6C727E"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "#F7F7F8",
          "borderRadius": 8.571428571428571,
          "flex": 1,
          "paddingHorizontal": 10.714285714285714,
          "paddingVertical": 15.694117647058823,
          "width": "100%",
        }
      }
      textAlignVertical="top"
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "paddingLeft": 21.428571428571427,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 28,
              },
              Array [
                undefined,
                Object {
                  "color": "#006C84",
                },
              ],
              Object {
                "fontFamily": "icomoon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          î¤Š
        </Text>
      </View>
    </View>
  </View>
</View>
`;
